src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/takemanagement/viewmodel/RecordableViewModel.kt

```diff
@@ -60,28 +65,30 @@ open class RecordableViewModel(private val audioPluginViewModel: AudioPluginView
         } ?: throw IllegalStateException("Recordable is null")
     }
 
-    fun editTake(take: Take) {
+    fun editTake(editTakeEvent: EditTakeEvent) {
         contextProperty.set(TakeContext.EDIT_TAKES)
         showPluginActive = true
         audioPluginViewModel
-            .edit(take)
+            .edit(editTakeEvent.take)
             .observeOnFx()
             .subscribe { result ->
                 showPluginActive = false
                 when (result) {
                     EditTake.Result.NO_EDITOR -> snackBarObservable.onNext(messages["noEditor"])
-                    EditTake.Result.SUCCESS -> {}
+                    EditTake.Result.SUCCESS -> editTakeEvent.onComplete()
                     null -> {} // This cannot happen but the compiler complains if null branch does not exist
```

Comment:
FYI, if you specify the type of result explicitly as non-null, this branch is unneeded.  I have no problem with this as-is though.

Reply:
Thanks! Great point.